apis/apiextensions/fn/io/v1alpha1/functionio_types.go

```diff
@@ -227,6 +230,46 @@ type DesiredReadinessCheck struct {
 	// type.
 	// +optional
 	MatchInteger *int64 `json:"matchInteger,omitempty"`
+
+	// MatchCondition specifies the condition you'd like to match if you're using "MatchCondition" type.
+	// +optional
+	MatchCondition *MatchConditionReadinessCheck `json:"matchCondition,omitempty"`
+}
+
+// MatchConditionReadinessCheck is used to indicate how to tell whether a resource is ready
+// for consumption
+type MatchConditionReadinessCheck struct {
+	// Type indicates the type of condition you'd like to use.
+	// +kubebuilder:default="Ready"
+	Type string `json:"type,omitempty"`
+
+	// Status is the status of the condition you'd like to match.
+	// +kubebuilder:default="True"
+	Status string `json:"status,omitempty"`
+}
+
+// DesiredReadinessCheckToV1 converts a DesiredReadinessCheck to a v1.ReadinessCheck.
+func DesiredReadinessCheckToV1(rc *DesiredReadinessCheck) (c *v1.ReadinessCheck) {
+	c.Type = v1.ReadinessCheckType(rc.Type)
+	if rc.FieldPath != nil {
+		c.FieldPath = *rc.FieldPath
+	}
+
+	// NOTE(negz): ComposedTemplate doesn't use pointer values for optional
+	// strings, so today the empty string and 0 are equivalent to "unset".
+	if rc.MatchString != nil {
+		c.MatchString = *rc.MatchString
+	}
+	if rc.MatchInteger != nil {
+		c.MatchInteger = *rc.MatchInteger
+	}
+	if rc.MatchCondition != nil {
+		c.MatchCondition = &v1.MatchConditionReadinessCheck{
+			Type:   rc.MatchCondition.Type,
+			Status: rc.MatchCondition.Status,
+		}
+	}
+	return c
```

Comment:
Is this change necessary? I generally prefer to avoid one package of user-facing API types importing another package of user-facing API types where possible. Largely to avoid a "slippery slope" that couples the two APIs more strongly - i.e. adding fields to this API whose types are imported from `apiextensions/v1`.

Would it be possible to keep the existing pattern from `ready.go` - i.e. translate both user-facing types to an internal type?

Reply:
I inverted the dependency moving the conversion code to be alongside the internal representation package, I would have probably preferred have some "ToInternal" methods, just to keep this logic nearer to the actual struct, so that if someone adds a new feature can see it needs to be propagated to the internal representation too, but that would cause a dependency cycle. That was the original reason to use v1 as intermediate representation